src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -3677,22 +3679,26 @@ UniqueFragmentList(List *fragmentList)
 
 	foreach(fragmentCell, fragmentList)
 	{
-		uint64 *shardId = NULL;
+		ShardInterval *shardInterval = NULL;
 		bool shardIdAlreadyAdded = false;
 		ListCell *uniqueFragmentCell = NULL;
 
 		RangeTableFragment *fragment = (RangeTableFragment *) lfirst(fragmentCell);
 		Assert(fragment->fragmentType == CITUS_RTE_RELATION);
 
-		shardId = fragment->fragmentReference;
+		Assert(CitusIsA(fragment->fragmentReference, ShardInterval));
+		shardInterval = (ShardInterval *) fragment->fragmentReference;
 
 		foreach(uniqueFragmentCell, uniqueFragmentList)
 		{
 			RangeTableFragment *uniqueFragment =
 				(RangeTableFragment *) lfirst(uniqueFragmentCell);
-			uint64 *uniqueShardId = uniqueFragment->fragmentReference;
+			ShardInterval *uniqueShardInterval = NULL;
+
+			Assert(CitusIsA(uniqueFragment->fragmentReference, ShardInterval));
```

Comment:
I added most of these because I thought it'd be possible that there's some in development code out there that passes around uint64*'s and these asserts will catch that.